osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimePlayer.cs

```diff
@@ -55,6 +62,31 @@ private void load()
                     this.Exit();
                 });
             }
+
+            Debug.Assert(client.Room != null);
+
+            int[] userIds = client.Room.Users.Where(u => u.State >= MultiplayerUserState.WaitingForLoad).Select(u => u.UserID).ToArray();
```

Comment:
Something to keep in mind is that this is unsafe given it's being done on the BDL thread while there are potentially users joining/leaving scheduled to the update thread. It could lead to collection modified during enumeration exceptions. I think the best way to do this is to pass in the userIds via ctor, which is guaranteed to be on the update thread.

Reply:
yeah, wasn't 100% on this, will move it.